cupyx/scipy/sparse/csr.py

```diff
@@ -357,3 +372,223 @@ def isspmatrix_csr(x):
 
     """
     return isinstance(x, csr_matrix)
+
+
+def multiply_by_scalar(sp, a):
+    data = sp.data * a
+    indices = sp.indices.copy()
+    indptr = sp.indptr.copy()
+    return csr_matrix((data, indices, indptr), shape=sp.shape)
+
+
+def multiply_by_dense(sp, dn):
+    sp_m, sp_n = sp.shape
+    dn_m, dn_n = dn.shape
+    if not (sp_m == dn_m or sp_m == 1 or dn_m == 1):
+        raise ValueError('inconsistent shape')
+    if not (sp_n == dn_n or sp_n == 1 or dn_n == 1):
+        raise ValueError('inconsistent shape')
+    m, n = max(sp_m, dn_m), max(sp_n, dn_n)
+    nnz = sp.nnz * (m // sp_m) * (n // sp_n)
+    dtype = functools.reduce(numpy.promote_types, (sp.dtype, dn.dtype))
```

Comment:
why not just 
`dtype = numpy.promote_types(sp.dtype, dn.dtype)`?

Reply:
Thank you, that's better.